### DISCUSSION TITLE
Log hp_metric only if improved

### DISCUSSION THREAD

**Discussion by lukasfolle** [Lightning-AI/pytorch-lightning#14385] [Idea pool]
Hi 👋 

I like to use Tensorboard to log my runs and compare different hyperparameter configurations. Sorting by hp_metric help identifying good runs, but since the last hp_metric might not be the best of a given run I would like to only log it if the metric acutally improved.

I couldn't find out how to access past stored metrics using `self.logger.experiment` but this should be possible right?

Then I could do:

```python
if past_hp_metric < current_hp_metric:  # max
    self.log("hp_metric", current_hp_metric)
```
